controller/array_action/array_mediator_svc.py

```diff
@@ -549,19 +548,20 @@ def _rollback_create_snapshot(self, target_volume_name):
         target_cli_volume = self._delete_unstarted_fcmap_if_exists(target_volume_name)
         self._delete_target_volume_if_exists(target_cli_volume)
 
-    def _create_snapshot(self, target_volume_name, source_volume_name):
+    def _create_snapshot(self, target_volume_name, source_volume_name, pool_id):
         try:
-            self._create_similar_volume(source_volume_name, target_volume_name)
+            self._create_similar_volume(source_volume_name, target_volume_name, pool_id)
             return self._create_and_start_fcmap(source_volume_name, target_volume_name, is_copy=False)
         except (svc_errors.CommandExecutionError, CLIFailureError) as ex:
             logger.error("Failed to create snapshot '{0}': {1}".format(target_volume_name, ex))
             logger.info("rolling back create snapshot '{0}'".format(target_volume_name))
             self._rollback_create_snapshot(target_volume_name)
             raise ex
 
-    def create_snapshot(self, name, volume_name, pool_id=None):
-        logger.info("creating snapshot '{0}' from volume '{1}'".format(name, volume_name))
-        target_cli_volume = self._create_snapshot(name, volume_name)
+    def create_snapshot(self, name, volume_id, pool_id):
+        logger.info("creating snapshot '{0}' from volume '{1}'".format(name, volume_id))
+        volume_name = self._get_volume_name_by_wwn(volume_id)
```

Comment:
consider changing this to `_get_cli_volume_by_wwn` and pass the `source_cli_volume` to `_create_snapshot` (instead of passing `volume_name`)

Reply:
done